File: Kadane's Algorithm/code.py
def all_max_subarrays(nums):
    max_ending_here = nums[0]
    max_so_far = nums[0]
    start = 0
    result = [[nums[0]]]

    for i in range(1, len(nums)):
        if nums[i] > max_ending_here + nums[i]:
            max_ending_here = nums[i]
            start = i
        else:
            max_ending_here += nums[i]

        if max_ending_here > max_so_far:
            max_so_far = max_ending_here
            result = [nums[start:i+1]]
        elif max_ending_here == max_so_far:
            result.append(nums[start:i+1])

    return max_so_far, result

# Example usage
arr = [-2, 1, -3, 4, -1, 2, 1, -5, 4]
max_sum, subarrays = all_max_subarrays(arr)

print("Max Sum:", max_sum)
print("All Max Subarrays:")
for sub in subarrays:
    print(sub)

File: Kadane's Algorithm/code.js
function allMaxSubarrays(nums) {
    let maxEndingHere = nums[0];
    let maxSoFar = nums[0];
    let start = 0;
    let result = [[nums[0]]];

    for (let i = 1; i < nums.length; i++) {
        if (nums[i] > maxEndingHere + nums[i]) {
            maxEndingHere = nums[i];
            start = i;
        } else {
            maxEndingHere += nums[i];
        }

        if (maxEndingHere > maxSoFar) {
            maxSoFar = maxEndingHere;
            result = [nums.slice(start, i + 1)];
        } else if (maxEndingHere === maxSoFar) {
            result.push(nums.slice(start, i + 1));
        }
    }

    return { maxSum: maxSoFar, subarrays: result };
}

// Example usage
const arr = [-2, 1, -3, 4, -1, 2, 1, -5, 4];
const result = allMaxSubarrays(arr);

console.log("Max Sum:", result.maxSum);
console.log("All Max Subarrays:");
result.subarrays.forEach(sub => console.log(sub));
